File: demo/src/main/java/com/example/demo/Question.java
package com.example.demo;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "Questions")
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "question_set_id", nullable = false)
    private Integer questionSetId;

    // "text" là tên cột trong DB; field đặt là questionText cho clear
    @Column(name = "text", nullable = false, columnDefinition = "NVARCHAR(MAX)")
    private String questionText;

    // "type" là tên cột trong DB
    @Column(name = "type", nullable = false, length = 20)
    private String type; // 'single' | 'multiple' | 'text'

    @Column(name = "order_in_set", nullable = false)
    private Integer orderInSet;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    // getters/setters
    public Long getId() { return id; }
    public Integer getQuestionSetId() { return questionSetId; }
    public void setQuestionSetId(Integer v) { this.questionSetId = v; }
    public String getQuestionText() { return questionText; }
    public void setQuestionText(String v) { this.questionText = v; }
    public String getType() { return type; }
    public void setType(String v) { this.type = v; }
    public Integer getOrderInSet() { return orderInSet; }

    public void setOrderInSet(Integer v) { this.orderInSet = v; }
    public Boolean getIsActive() { return isActive; }
    public void setIsActive(Boolean v) { this.isActive = v; }
    
}

File: frontend/lib/Admin/list_of_questions/questions/question.dart
// models/question.dart
class Question {
  final int id;
  String text;
  final int setId; // question_set_id từ DB
  String setName; // map từ setId -> tên bộ (để UI cũ chạy)
  int order; // order_in_set
  bool isActive;
  final String? type; // 'single' | 'multiple' | 'text' (nếu cần)

  Question({
    required this.id,
    required this.text,
    required this.setId,
    required this.setName,
    required this.order,
    required this.isActive,
    this.type,
  });

  Question copy({
    int? id,
    String? text,
    int? setId,
    String? setName,
    int? order,
    bool? isActive,
    String? type,
  }) {
    return Question(
      id: id ?? this.id,
      text: text ?? this.text,
      setId: setId ?? this.setId,
      setName: setName ?? this.setName,
      order: order ?? this.order,
      isActive: isActive ?? this.isActive,
      type: type ?? this.type,
    );
  }

  // Nếu bạn đã có map setId->name thì truyền vào để dựng setName
  factory Question.fromJson(Map<String, dynamic> j,
      {String Function(int)? setNameOf}) {
    final setId = j['questionSetId'] ?? j['question_set_id'];
    final name = (setNameOf != null) ? setNameOf(setId) : 'Set #$setId';
    return Question(
      id: j['id'] as int,
      text: j['questionText'] ?? j['text'],
      setId: setId as int,
      setName: name,
      order: j['orderInSet'] ?? j['order_in_set'] as int,
      isActive: (j['isActive'] ?? j['is_active'] ?? true) as bool,
      type: j['type'] as String?,
    );
  }
}
